### PR TITLE
Trq/gpu multicopy

### PR DESCRIPTION
**Pull request by trquinn**
Make the local data transfer to GPU faster by using multiple threads to copy the data into the buffer.hh

### CODE REVIEW

File: DataManager.h [hunk 1]
```diff
@@ -104,6 +104,7 @@ class DataManager : public CBase_DataManager {
         // * either do not concern yourself with cached particles
         // * or for each entry, get key, find bucket node in CM, DM or TPs and get number
         // for now, the former
+	int treePiecesBufferFilled;
```
Comment (trquinn) [inline DataManager.h:107]: This line has been inserted between a comment (above) and the variable it is describing (below).  This should be moved and also commented.

File: DataManager.h [hunk 2]
```diff
@@ -130,6 +131,8 @@ class DataManager : public CBase_DataManager {
         CompactPartData *bufRemoteParts;
 
         /// host buffer to transfer local moments to GPU
+        CkVec<CudaMultipoleMoments> localMoments;
+        CkVec<CompactPartData> localParticles;
```
Comment (trquinn) [inline DataManager.h:135]: Unused!

Comment (trquinn) [reply DataManager.h]: Now deleted.

**Comment by trquinn** [comment]
N.B.  My last commit was only tested on my laptop.  It needs to be tested on an actual GPU.

**Comment by spencerw** [comment]
I'm realizing that all of the tests I've run so far have been with -DHAPI_TRACE enabled. Currently, this fails to build unless that flag is enabled. The CI doesn't catch this since it only does a non-cuda build.

On a related note, I'm finding that 'testenergy' fails if HAPI_TRACE is disabled. I tried out a few different commits, and this issue extends at least back to the beginning of September before we started the project. Building everything in non-SMP mode, however, works OK.

**Review by spencerw — COMMENTED** [review]
Missing #ifdef HAPI_TRACE

File: DataManager.cpp [hunk 3]
```diff
@@ -849,50 +843,12 @@ void DataManager::serializeLocal(GenericTreeNode *node){
     }
   }// end while queue not empty
 
+  traceUserBracketEvent(SER_LOCAL_WALK, starttime, CmiWallTimer());
```
[the remark this answers is not in the document]
Comment (trquinn) [reply DataManager.cpp:847]: fixed in latest commit

**Pull request merged**